### PR TITLE
Update docs with linux profile

### PR DESCRIPTION
**Pull request by jbescos**
Relates to https://github.com/eclipse-ee4j/mail/pull/539

### CODE REVIEW

File: Mbox-Store.md [hunk 1]
```diff
@@ -62,7 +62,13 @@ your PATH and the JDK is in /usr/java:
     cd mbox
     mvn
     cd native
-    mvn
+    mvn -P1.7
```
Comment (lukasj) [inline Mbox-Store.md:65]: 1.7 profile should not be required

Comment (jbescos) [reply Mbox-Store.md:65]: Currently  it is necessary. There is one validation that forces to use JDK11 if I don't specify the profile:
```
[INFO] --- maven-enforcer-plugin:3.0.0-M3:enforce (enforce-version) @ libmbox ---
[WARNING] Rule 1: org.apache.maven.plugins.enforcer.RequireJavaVersion failed with message:
Detected JDK Version: 1.8.0-231 is not in the allowed range [11,).
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.899 s
[INFO] Finished at: 2021-03-26T12:07:26+01:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal org.apache.maven.plugins:maven-enforcer-plugin:3.0.0-M3:enforce (enforce-version) on project libmbox: Some Enforcer rules have failed. Look above for specific messages explaining why the rule failed. -> [Help 1]
```

I will need to modify a pom.xml file to remove this requirement.

Comment (lukasj) [reply Mbox-Store.md:65]: jdk 11 for build is required

Comment (lukasj) [reply Mbox-Store.md:65]: in the other words, the build is supposed to work also on JDK 11

Comment (jbescos) [reply Mbox-Store.md:65]: I have updated the doc. The JDK 11 fix is in this other PR:
https://github.com/eclipse-ee4j/mail/pull/541

File: Mbox-Provider.md [hunk 2]
```diff
@@ -5,19 +5,26 @@ The mbox provider is a Jakarta Mail local store provider that manages files
...
-To build the mbox provider:
+To build the mbox provider for Solaris/OpenSolaris:
 
     % (cd mbox; mvn)
     % (cd mbox/native; mvn)
 
+To build the mbox provider for distributions based on Debian:
+
+    % (cd mbox; mvn)
+    % (cd mbox/native; mvn linux)
```
Comment (pzygielo) [inline Mbox-Provider.md:22]:
```
[ERROR] Unknown lifecycle phase "linux"
```

File: Mbox-Store.md [hunk 3]
```diff
@@ -64,6 +64,12 @@ your PATH and the JDK is in /usr/java:
     cd native
     mvn
 
+In distributions based on Debian you need to to have installed
+liblockfile-dev package and use the profile **linux** when executing
+mvn in the native folder:
+
+    mvn linux
```
Comment (pzygielo) [inline Mbox-Store.md:71]:
```
[ERROR] Unknown lifecycle phase "linux"
```

File: Mbox-Provider.md (hunk 2 above)
Comment (jbescos) [reply Mbox-Provider.md]: Thank you for pointing me that. It is with -P

**Review by lukasj — APPROVED** [review]

**Pull request merged**